### PR TITLE
131 suggestions

### PR DESCRIPTION
**Pull request by eduardb**

### CODE REVIEW

File: app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt [hunk 1]
```diff
@@ -32,7 +36,7 @@ class MaskCustomView @JvmOverloads constructor(
     var onMaskingActionFinishedListener: OnTouchListener? = null
     private val maskLayer = MaskLayer(context, imageMatrix)
     private var currentMode: Mode = Mode.Draw
-    private lateinit var drawableDimensions: Pair<Int, Int>
+    private lateinit var drawableDimensions: Size
```
Comment (raul-oliver-tw) [inline app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt:39]: Do you mind renaming this to drawableSize, please?

Comment (eduardb) [reply app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt]: Good catch! :) Fixed.

File: app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt [hunk 2]
```diff
@@ -57,13 +61,13 @@ class MaskCustomView @JvmOverloads constructor(
...
     private fun onTouchDraw(event: MotionEvent): Boolean {
-        val (x, y) = invert(event.x, event.y)
-        val (drawableWidth, drawableHeight) = drawableDimensions
         if (event.action == MotionEvent.ACTION_UP) {
             maskLayer.drawEnd()
             onMaskingActionFinishedListener?.onTouch(this, event)
             invalidate()
         } else {
+            val (x, y) = invert(event.x, event.y)
+            val (drawableWidth, drawableHeight) = drawableSize
```
Comment (eduardb) [inline app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt:70]: I also moved these closer to where they are used.

File: app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt (hunk 1 above)
Comment (raul-oliver-tw) [reply app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt]: Great! Thanks!!

File: app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt (hunk 2 above)
Comment (raul-oliver-tw) [reply app/src/main/java/net/aiscope/gdd_app/ui/mask/customview/MaskCustomView.kt:70]: Nice! Thanks!!

**Pull request merged**